crates/sui-move/src/build.rs

```diff
@@ -96,3 +108,42 @@ pub fn resolve_lock_file_path(
     build_config.lock_file = Some(lock_file_path);
     Ok(build_config)
 }
+
+pub const PUBLISHED_AT_MANIFEST_FIELD: &str = "published-at";
+
+pub struct SuiPackageHooks {}
+
+impl PackageHooks for SuiPackageHooks {
+    fn custom_package_info_fields(&self) -> Vec<String> {
+        vec![PUBLISHED_AT_MANIFEST_FIELD.to_string()]
+    }
+
+    fn custom_dependency_key(&self) -> Option<String> {
+        None
+    }
+
+    fn resolve_custom_dependency(
+        &self,
+        _dep_name: move_symbol_pool::Symbol,
+        _info: &CustomDepInfo,
+    ) -> anyhow::Result<()> {
+        Ok(())
+    }
+}
+
+pub fn resolve_manifest_lookup(
+    package_path: Option<PathBuf>,
+) -> anyhow::Result<BTreeMap<String, String>> {
+    let package_root = base::reroot_path(package_path)?;
+    let manifest_string =
+        std::fs::read_to_string(package_root.join(layout::SourcePackageLayout::Manifest.path()))?;
+    let toml_manifest = manifest_parser::parse_move_manifest_string(manifest_string)?;
+    let manifest = manifest_parser::parse_source_manifest(toml_manifest)?;
```

Comment:
There's a function in `move_package::source_package::manifest_parser` that should do all these things for you:

```suggestion
    let manifest = manifest_parser::parse_move_manifest_from_file(package_root)?;
```